File: fresh-shop/server/api/v1/business/bill.go
package business

import (
	"fresh-shop/server/middleware"
	"fresh-shop/server/model/common/response"
	"fresh-shop/server/service/business"
	"github.com/gin-gonic/gin"
)

// BillApi 账单API
type BillApi struct{}

var billService = &business.BillService{}

// GetBillList 获取账单列表
// GET /api/business/bill/list?page=1&pageSize=10
func (api *BillApi) GetBillList(c *gin.Context) {
	var req struct {
		Page     int `form:"page"`
		PageSize int `form:"pageSize"`
	}
	if err := c.ShouldBindQuery(&req); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}

	companyId := middleware.GetCompanyID(c)
	if companyId == 0 {
		response.FailWithMessage("无法获取公司信息", c)
		return
	}

	list, total := billService.GetBillList(companyId, req.Page, req.PageSize)
	response.OkWithDetailed(gin.H{
		"list":  list,
		"total": total,
	}, "查询成功", c)
}

// GenerateBill 生成账单
// POST /api/business/bill/generate
// Body: { "year": 2026, "month": 9 }
func (api *BillApi) GenerateBill(c *gin.Context) {
	var req struct {
		Year  int `json:"year" binding:"required"`
		Month int `json:"month" binding:"required"`
	}
	if err := c.ShouldBindJSON(&req); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}

	companyId := middleware.GetCompanyID(c)
	if companyId == 0 {
		response.FailWithMessage("无法获取公司信息", c)
		return
	}

	if err := billService.GenerateBill(companyId, req.Year, req.Month); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	response.OkWithMessage("账单生成成功", c)
}

// GetBillDetail 获取账单详情
// GET /api/business/bill/:id
func (api *BillApi) GetBillDetail(c *gin.Context) {
	var req struct {
		ID uint `uri:"id" binding:"required"`
	}
	if err := c.ShouldBindUri(&req); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}

	bill, err := billService.GetBillDetail(req.ID)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	response.OkWithData(bill, "查询成功", c)
}

// UpdateBillStatus 更新账单状态
// PUT /api/business/bill/:id/status
func (api *BillApi) UpdateBillStatus(c *gin.Context) {
	var req struct {
		ID     uint `uri:"id" binding:"required"`
		Status int  `json:"status" binding:"required"`
	}
	if err := c.ShouldBindUri(&req); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	if err := c.ShouldBindJSON(&req); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}

	if err := billService.UpdateBillStatus(req.ID, req.Status); err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	response.OkWithMessage("状态更新成功", c)
}

File: fresh-shop/web/src/api/bill.js
import service from '@/utils/request'

// @Tags Bill
// @Summary 获取账单列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query { page: 1, pageSize: 10 } true "分页参数"
// @Success 200 {string} string "{"success":true,"data":{"list":[],"total":0},"msg":"查询成功"}"
// @Router /business/bill/list [get]
export const getBillList = (params) => {
  return service({
    url: '/business/bill/list',
    method: 'get',
    params
  })
}

// @Tags Bill
// @Summary 生成账单
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body { year: 2026, month: 9 } true "账单年月"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"账单生成成功"}"
// @Router /business/bill/generate [post]
export const generateBill = (data) => {
  return service({
    url: '/business/bill/generate',
    method: 'post',
    data
  })
}

// @Tags Bill
// @Summary 获取账单详情
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /business/bill/:id [get]
export const getBillDetail = (id) => {
  return service({
    url: `/business/bill/${id}`,
    method: 'get'
  })
}

// @Tags Bill
// @Summary 更新账单状态
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Success 200 {string} string "{"success":true,"data":{},"msg":"状态更新成功"}"
// @Router /business/bill/:id/status [put]
export const updateBillStatus = (id, status) => {
  return service({
    url: `/business/bill/${id}/status`,
    method: 'put',
    data: { status }
  })
}
